### PR TITLE
Add a travis suite to run the staging store tests

### PR DESCRIPTION
**Pull request by elopio**
LP: #1596072

### CODE REVIEW

**Comment by elopio** [comment]
skip ci

**Comment by snappy-m-o** [comment]
Test jenkins

**Comment by elopio** [comment]
This is good to go. As good as it gets at least. I launched the travis tests many many times, and in the end found that it will fail if the two executions sync and run a registration within 5 seconds. There's nothing we can do about it right now. We could sign up a different user for each test, but there is no way to accept the store agreement with a script.
The error happened like one every 10 times, so a retrigger would get us to green. And in addition to that, this will happen only when the branch is in snapcore/snapcraft, which we don't do often. And then the execution travis does for new pushes to master is a single one, and unlikely to sync up with another run against staging.

**Comment by elopio** [comment]
retest this please

**Comment by elopio** [comment]
retest this please

File: integration_tests/__init__.py [hunk 1]
```diff
@@ -111,3 +112,9 @@ def logout(self):
         expected = ('Clearing credentials for Ubuntu One SSO.\n'
                     'Credentials cleared.\n')
         self.assertEqual(expected, output)
+
+    def register(self, snap_name):
+        self.run_snapcraft(['register', snap_name])
+        # sleep a few seconds to avoid hitting the store restriction on
+        # following registrations.
+        time.sleep(10)
```
Comment (celso-providelo) [inline integration_tests/__init__.py]: For the record, Store staging has 5 s rate-limit and production 60 s, a comment on the test would not hurt.

**Comment by elopio** [comment]
@celso-providelo I didn't use the specific rate, and didn't comment much about it because you told me you were not comfortable making that info public :p 
I will have to come back and refactor this when I enable the tests to run in production. I can add more comments then.

**Comment by celso-providelo** [comment]
@elopio slow-clapping for me, huh ?! As we discussed before, when this delay starts impacting the full-suite run we will have to find a better way (no rate-limit for specific users/groups).

**Pull request merged**